Feature/initial code cb 6812

CombeeMike opened the pull request:
[Paymo Task](https://app.paymoapp.com/#Paymo.Projects/2013095/tasks/simple/task/22169148)

Shared ESLint config which is published to new NPM package [@combeenation/eslint-config](https://www.npmjs.com/package/@combeenation/eslint-config).
The config is based on the existing one from CJS utils augmented with some personal preferences which I'm happy to discuss with you 2.

Most important changes to our currently used configs which make the linter a bit more strict:
* `no-use-before-define`
* `@typescript-eslint/explicit-module-boundary-types`: This basically requires explicit annotation of function return types even if TS could infer them.
  I personally like this explicitness but I also see that it can be cumbersome at some times and am therefore very interested in hearing your opinions on that.
  This one yields many new warnings when applied to our current rebuild code base.

Usage see other PRs.

CombeeMike committed:
Setup code and initial implementation

#CB-6812

CombeeMike committed:
0.0.1-rc1

CombeeMike committed:
Add npm publish scripts

#CB-6812

CombeeMike committed:
Install published cbn-prettier-config

#CB-6812

CombeeMike committed:
0.0.1-rc2

HoferMarkus commented on index.js:
```diff
@@ -0,0 +1,36 @@
+    // We sometimes use `let` instead of `const` even when ESLint "thinks" that `const` would be possible.
+    // E.g. when initializing a variable with an array and mutating the array contents later on, this is technically
+    // still a const but we tend to use `let` in such situations as that is IMHO more "semantically correct".
+    'prefer-const': 'off',
```
Maybe we could rethink that in some time.
Even if `let` seems to be easier to understand in some cases it's the official JavaScript way to use `const` in more scenarios.
Personally I also think that it would be an approvement, if I don't have to think about `const` vs `let` so much, because the linter does that job for me.

CombeeMike replied:
I'm ok with that as well. If @cbn-falias also approves, we can IMHO change that rule right away...

cbn-falias replied:
I don't really have a preference on that, so I would also let the linter do the job. 👍

CombeeMike replied:
FYI, removed the rule -> defaults now to `'prefer-const': 'on'`.

HoferMarkus reviewed (approved):
LGTM

cbn-falias reviewed (approved):
LGTM

CombeeMike committed:
Add correct URL to (now) public repo

#CB-6812

CombeeMike committed:
Reset "prefer-const" to (inherited) default

See PR discussion for details:
https://github.com/Combeenation/cbn-eslint-config/pull/1#discussion_r822342070

#CB-6812

CombeeMike committed:
0.0.1
